### PR TITLE
Required signatures on protected branches

### PR DESCRIPTION
**Pull request by danijeel**
Added support for the preview required commit signatures for Protected Branches API.

https://developer.github.com/changes/2018-02-22-protected-branches-required-signatures/

This solves: https://github.com/google/go-github/issues/902

### CODE REVIEW

File: github/repos.go [hunk 1]
```diff
@@ -850,6 +857,67 @@ func (s *RepositoriesService) RemoveBranchProtection(ctx context.Context, owner,
...
+	p := new(SignaturesProtectedBranch)
+	resp, err := s.client.Do(ctx, req, p)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return p, resp, nil
+}
+
+// AddSignatureProtectedBranch Require signed commits to a protected branch.
```
Comment (gmlewis) [inline github/repos.go]: I'm wondering if this should be renamed... I'm not even happy with the working on the GitHub API docs themselves.
From the description at the provided URL, it sounds like this endpoint is used to require signatures on protected branches.  So how about this:

`// RequireSignaturesOnProtectedBranch makes signed commits required on a protected branch.```

File: github/repos.go [hunk 2]
```diff
@@ -850,6 +857,67 @@ func (s *RepositoriesService) RemoveBranchProtection(ctx context.Context, owner,
...
+	r := new(SignaturesProtectedBranch)
+	resp, err := s.client.Do(ctx, req, r)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return r, resp, err
+}
+
+// RemoveSignatureProtectedBranch removes required signed commits on a given branch.
```
Comment (gmlewis) [inline github/repos.go]: Maybe `OptionalSignaturesOnProtectedBranch` ?

**Comment by danijeel** [comment]
Thank you for the review and feedback @gmlewis! I've updated the comments you mention above.

File: github/repos.go [hunk 3]
```diff
@@ -698,11 +698,11 @@ type DismissalRestrictionsRequest struct {
 	Teams *[]string `json:"teams,omitempty"`
 }
 
-// SignaturesProtectedBranch represents the protection status of a individual branch.
+// SignaturesProtectedBranch represents the protection status of an individual branch.
 type SignaturesProtectedBranch struct {
 	URL *string `json:"url,omitempty"`
 	// Commits pushed to matching branches must have verified signatures.
-	Enabled bool `json:"strict"`
+	Enabled bool `json:"strict,omitempty"`
```
Comment (gmlewis) [inline github/repos.go]: This still needs `*bool`.

File: github/repos_test.go [hunk 4]
```diff
@@ -1066,13 +1066,13 @@ func TestRepositoriesService_GetSignaturesProtectedBranch(t *testing.T) {
 	})
 
 	signature, _, err := client.Repositories.GetSignaturesProtectedBranch(context.Background(), "o", "r", "b")
+
```
Comment (gmlewis) [inline github/repos_test.go]: Please delete this blank line, as we attempt to keep the error checking code tightly bound to the section of code that created it.

File: github/repos_test.go [hunk 5]
```diff
@@ -1066,13 +1066,13 @@ func TestRepositoriesService_GetSignaturesProtectedBranch(t *testing.T) {
...
 	if err != nil {
 		t.Errorf("Repositories.GetSignaturesProtectedBranch returned error: %v", err)
 	}
 
 	want := &SignaturesProtectedBranch{
-		URL:     String("/repos/o/r/branches/b/protection/required_signatures"),
-		Enabled: false,
+		URL: String("/repos/o/r/branches/b/protection/required_signatures"),
```
Comment (gmlewis) [inline github/repos_test.go]: Please add back your `Enabled: Bool(false),` line here and below.

File: github/repos_test.go [hunk 6]
```diff
@@ -1096,9 +1096,9 @@ func TestRepositoriesService_AddSignatureProtectedBranch(t *testing.T) {
...
```
Comment (gmlewis) [inline github/repos_test.go]: Please add back your `Enabled: Bool(false),` line here.

File: github/repos_test.go [hunk 7]
```diff
@@ -1055,6 +1055,72 @@ func TestRepositoriesService_RemoveAdminEnforcement(t *testing.T) {
...
+		URL:     String("/repos/o/r/branches/b/protection/required_signatures"),
+		Enabled: Bool(false),
+	}
+
+	if !reflect.DeepEqual(signature, want) {
+		t.Errorf("Repositories.RequireSignaturesOnProtectedBranch returned %+v, want %+v", signature, want)
+	}
+}
+
+func TestRepositoriesService_RemoveSignatureProtectedBranch(t *testing.T) {
```
Comment (gmlewis) [inline github/repos_test.go]: s/RemoveSignature/OptionalSignaturesOn/

File: github/repos.go [hunk 8]
```diff
@@ -698,6 +698,13 @@ type DismissalRestrictionsRequest struct {
 	Teams *[]string `json:"teams,omitempty"`
 }
 
+// SignaturesProtectedBranch represents the protection status of an individual branch.
+type SignaturesProtectedBranch struct {
+	URL *string `json:"url,omitempty"`
+	// Commits pushed to matching branches must have verified signatures.
+	Enabled *bool `json:"strict,omitempty"`
```
Comment (gmlewis) [inline github/repos.go]: s/strict/enabled/

**Pull request merged**